backend/src/main/java/ro/utcn/sd/mdantonio/StackUnderflow/controller/PostController.java

```diff
@@ -0,0 +1,89 @@
+package ro.utcn.sd.mdantonio.StackUnderflow.controller;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.*;
+import ro.utcn.sd.mdantonio.StackUnderflow.dto.PostDTO;
+import ro.utcn.sd.mdantonio.StackUnderflow.dto.VoteDTO;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.PostManagementService;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.TagManagementService;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.UserSessionManagementService;
+
+import java.util.List;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+public class PostController {
+    private final PostManagementService postManagementService;
+    private final TagManagementService tagManagementService;
+    private final UserSessionManagementService userSessionManagementService;
+
+    @GetMapping("/posts")
+    public List<PostDTO> readAllQuestions() {
+        return postManagementService.listQuestions();
+    }
+
+
+    @PostMapping("/posts")
+    public PostDTO createPost(@RequestBody PostDTO dto) {
+        int currentUserId = userSessionManagementService.loadCurrentUser().getUserid();
+        int postType = PostDTO.convertStringPostTypeToIntPostType(dto.getType());
+        PostDTO response = postManagementService.addPost(postType, dto.getAuthor().getId(), dto.getParent(), dto.getTitle(), dto.getBody(), dto.getDate());
+        dto.getTags().forEach(tagId -> tagManagementService.attachTagToPost(currentUserId, tagId, response.getId()));
+        return postManagementService.findPostById(response.getId());
+    }
+
+
+    @PutMapping("/posts")
```

Comment:
Usually for update calls, you also have the ID in the url (like `/posts/{id}`), although it is somehow redundant due to the fact that you probably have it inside the body.